edb/schema/referencing.py

```diff
@@ -687,6 +686,41 @@ def _validate(
                     context=self.source_context,
                 )
 
+    def get_implicit_bases(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+        bases: Any,
+    ) -> Sequence[str]:
+
+        mcls = self.get_schema_metaclass()
+        default_base = mcls.get_default_base_name()
+
+        if isinstance(bases, so.ObjectCollectionShell):
+            base_names = [
+                b.name for b in bases.items if b.name is not None
```

Comment:
It's hard to trace the API while reviewing, so I'll just ask: why `.name` and not `.get_name(schema)`?

Reply:
It's not an object it's an `ObjectShell`, but it may be worth imitating the object API on it.

Reply:
+1 to imitate it, for sure.